### PR TITLE
Builds new library structure for OpenROAD modules

### PR DESCRIPTION
**Pull request by QuantamHD**
The new library ordering of modules decouples the TCL/Wrapper code from the underlying module code. This allows us to create unit tests for each module without having to link the entire application and swig layers.

It should lead to a more ~~compostable~~ composable environment.

And now a poem, because I like being extra.

```
The path is dark and cold,
The trees are gnarled and old,
The sun is hidden from my sight,
And I am lost in endless night.

I stumble on, I know not where,
My only guide is my despair.
The path is hard and the journey long,
And I am weak, I cannot go on.

But then I see a light,
A glimmer of hope in the night.
I run towards it, I must find it,
It is my only chance to be free.

I stumble out into the light,
And I see that I have died.
I am in a new world,
A world of peace and light.

I am free.
```

Signed-off-by: Ethan Mahintorabi <ethanmoon@google.com>

### CODE REVIEW

**Comment by QuantamHD** [comment]
@cdleary

**Comment by vvbandeira** [comment]
fyi CI is failing at pr-head because is missing https://github.com/The-OpenROAD-Project/OpenROAD/pull/2841
Merging master back into your branch should resolve the build issue.

**Comment by maliberty** [comment]
compostable :)

File: src/rcx/src/CMakeLists.txt [hunk 1]
```diff
@@ -70,7 +85,7 @@ target_include_directories(rcx
 
 target_link_libraries(rcx
```
Comment (maliberty) [inline src/rcx/src/CMakeLists.txt:86]: should rcx_py depend on rcx or rcx_lib?

File: src/rcx/include/rcx/ext.h [hunk 2]
```diff
@@ -56,7 +55,10 @@ class Ext
   Ext();
   ~Ext() = default;
 
-  void init(Tcl_Interp* tcl_interp, odb::dbDatabase* db, Logger* logger);
+  void init(
+      odb::dbDatabase* db,
```
Comment (maliberty) [inline src/rcx/include/rcx/ext.h:59]: clang-format

**Comment by QuantamHD** [comment]
> compostable :)

Thanks 👍

File: src/rcx/src/CMakeLists.txt (hunk 1 above)
Comment (QuantamHD) [reply src/rcx/src/CMakeLists.txt:86]: It seems like the py module wants the swig libs for something it doesn't work with the lib at this time. Any ideas?

Comment (maliberty) [reply src/rcx/src/CMakeLists.txt:86]: Not without trying it myself.  What sort of problem arises?

File: src/rcx/src/CMakeLists.txt [hunk 3]
```diff
@@ -33,34 +33,49 @@
...
+  PUBLIC
+    ../include
+)
+
+target_link_libraries(rcx_lib
+  PUBLIC
+    odb
+    utl
+  PRIVATE
+    OpenSTA
```
[the remark this answers is not in the document]
Comment (QuantamHD) [reply src/rcx/src/CMakeLists.txt]: Done.

File: src/rcx/src/CMakeLists.txt (hunk 1 above)
Comment (QuantamHD) [reply src/rcx/src/CMakeLists.txt:84]: Done. Fixed the issue.

**Review by maliberty — APPROVED** [review]

**Pull request merged**